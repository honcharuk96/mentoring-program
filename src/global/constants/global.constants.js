import { convertGenres, convertSelectedGenres } from '../../postersBlock/posterForm/formHelper';

export const statusForm = {
  ADD: 'ADD',
  UPDATE: 'UPDATE',
  DELETE: 'DELETE',
};

export const getDefaultPosterState = ( withId = false) => {
    const defData = {
      title: '',
      vote_average: 0,
      release_date: '',
      poster_path: '',
      runtime: '',
      overview: '',
      genres: convertGenres([]),
      selectedGenres: convertSelectedGenres([]),
    };
    return withId ? {id:'', ...defData} : defData
}

export const convertPosterState = (data, withId = false) => {
    const poster = {
      title: data.title,
      vote_average: +data.vote_average,
      release_date: data.release_date,
      poster_path: data.poster_path,
      runtime: +data.runtime,
      overview: data.overview,
      genres: convertSelectedGenres(data.selectedGenres, false),
    }
    return withId ? {id:data.id, ...poster} : poster
}
