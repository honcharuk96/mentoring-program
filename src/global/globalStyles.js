import styled, { createGlobalStyle } from 'styled-components';

import fontData from './fonts/fonts.constants';

const { MONTSERRAT } = fontData;
const GlobalStyles = createGlobalStyle`
  @font-face {
    font-family: ${MONTSERRAT.name};
    font-style: normal;
    font-weight: 400;
    src: url(${MONTSERRAT.eot}); /* IE9 Compat Modes */
    src: local(''),
    url('${MONTSERRAT.eot}?#iefix') format('embedded-opentype'), /* IE6-IE8 */
    url(${MONTSERRAT.woff2}) format('woff2'), /* Super Modern Browsers */ 
    url(${MONTSERRAT.woff}) format('woff'), /* Modern Browsers */ 
    url(${MONTSERRAT.ttf}) format('truetype'), /* Safari, Android, iOS */ 
    url('${MONTSERRAT.svg}#Montserrat') format('svg'); /* Legacy iOS */
  }

  * {
    font-family: ${MONTSERRAT.name}, Helvetica, Arial, Verdana, Tahoma, sans-serif;
    box-sizing: border-box;
  }

  html, body {
    height: 100%;
  }


  body {
    padding: 0;
    margin: 0;
    background: transparent;
    font-size: 16px;
    font-stretch: normal;
    font-style: normal;
    font-weight: normal;
    line-height: 1.2;
    letter-spacing: normal;
    overflow-x: hidden;
    overflow-y: auto;
  }

  #root {
    width: 100%;
    height: 100%;
  }
`;

export default GlobalStyles;

export const MainWrapper = styled.div`
  padding: 0 50px;
  background: #232323;
`;

export const Line = styled.div`
  height: 4px;
  border-bottom: 2px solid #080707;
  background: #424242;
`;

export const Bold = styled.span`
  font-weight: 900;
`;
