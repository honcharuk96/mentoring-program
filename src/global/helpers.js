export const convertMinsToHrsMins = mins => {
  const h = Math.floor(mins / 60);
  const m = mins % 60;
  return `${h}h ${m}min`;
};
