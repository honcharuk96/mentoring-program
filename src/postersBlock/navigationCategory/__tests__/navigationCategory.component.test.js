import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import NavigationCategory, { SortComponent } from '../navigationCategory.component';
import { links, variantSorts } from '../../../global/constants/global.constants';
import { screen } from '@testing-library/dom';
const changeActiveNav = jest.fn();
const changeVariantSort = jest.fn();
const getPostersByActiveNavWithSort = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: jest.fn(),
    location: {},
  }),
}));
describe('<NavigationCategory />', () => {
  it('should render NavigationCategory', () => {
    const { asFragment } = render(
      <NavigationCategory
        activeNav={links[0].text}
        changeActiveNav={changeActiveNav}
        changeVariantSort={changeVariantSort}
        getPostersByActiveNavWithSort={getPostersByActiveNavWithSort}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render SortComponent', () => {
    const { asFragment } = render(<SortComponent changeVariantSort={changeVariantSort} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should select correct time on change', () => {
    render(<SortComponent changeVariantSort={changeVariantSort} />);
    expect(screen.getByText(variantSorts[0].label)).toBeInTheDocument();

    fireEvent.change(screen.getByTestId('select'), {
      target: { value: variantSorts[1].name },
    });
    expect(screen.getByText(variantSorts[1].label)).toBeInTheDocument();
  });
});
