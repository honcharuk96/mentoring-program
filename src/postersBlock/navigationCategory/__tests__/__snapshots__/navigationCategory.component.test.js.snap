// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationCategory /> should render NavigationCategory 1`] = `
<DocumentFragment>
  <nav
    class="navigationCategorystyled__NavigationList-sc-1bdsphd-0 bfkBTY"
  >
    <li
      class="navigationCategorystyled__NavigationItem-sc-1bdsphd-1 cVQqkF"
    >
      all
    </li>
    <li
      class="navigationCategorystyled__NavigationItem-sc-1bdsphd-1 iaUwum"
    >
      documentary
    </li>
    <li
      class="navigationCategorystyled__NavigationItem-sc-1bdsphd-1 iaUwum"
    >
      comedy
    </li>
    <li
      class="navigationCategorystyled__NavigationItem-sc-1bdsphd-1 iaUwum"
    >
      horror
    </li>
    <li
      class="navigationCategorystyled__NavigationItem-sc-1bdsphd-1 iaUwum"
    >
      crime
    </li>
    <div
      class="navigationCategorystyled__SortListBlock-sc-1bdsphd-4 kAzxwq"
    >
      <div
        class="navigationCategorystyled__SortListText-sc-1bdsphd-3 ioMoJm"
      >
         Sort by
      </div>
      <select
        class="navigationCategorystyled__SortList-sc-1bdsphd-2 cAbzVg"
        data-testid="select"
        name="sort"
      >
        <option
          value="release_date"
        >
          Release date
        </option>
        <option
          value="vote_average"
        >
          Rating
        </option>
      </select>
    </div>
  </nav>
  <div
    class="globalStyles__Line-sc-osyo6d-1 fxqYku"
  />
</DocumentFragment>
`;

exports[`<NavigationCategory /> should render SortComponent 1`] = `
<DocumentFragment>
  <div
    class="navigationCategorystyled__SortListBlock-sc-1bdsphd-4 kAzxwq"
  >
    <div
      class="navigationCategorystyled__SortListText-sc-1bdsphd-3 ioMoJm"
    >
       Sort by
    </div>
    <select
      class="navigationCategorystyled__SortList-sc-1bdsphd-2 cAbzVg"
      data-testid="select"
      name="sort"
    >
      <option
        value="release_date"
      >
        Release date
      </option>
      <option
        value="vote_average"
      >
        Rating
      </option>
    </select>
  </div>
</DocumentFragment>
`;
